Assets/Scripts/Canvas/PropertiesMenuHandler.cs

```diff
@@ -48,16 +57,29 @@ public void Start()
 
     public void Update()
     {
+      if (_nodeClasses == null)
+      {
+        _graphService = FindObjectOfType<GraphService>();
```

Comment:
Any reason to refetch _graphService on every frame? Is there a possibility that the old one is somehow invalid?

Reply:
The problem is that it seems random when the objects are created by Unity. Of course I wanted to assign the node classes in the Start method, but graph service is null there. As a workaround it is done here, and the fetches will stop after a few frames when the node classes are loaded here. 
I hate this workaround but don't have a better idea for this. Ideally I would like to rehaul our scripts completely so it is deterministic when they are created (possibly some DI)

Reply:
You can kindof manually inject an object in editor. It's cumbersome and usually gets messed up in merges though

Reply:
Yeah but I feel like we are not using the editor assignements properly. Most of our scripts are singleton-like and it would be much easier to just inject them in the code

Reply:
Agreed. I don't really know what does editor assignments are for tbh, they don't scale at all